compiler/luci/pass/src/FoldShapePass.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FoldShapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+namespace
+{
+
+template <loco::DataType OutType> luci::CircleConst *folding(luci::CircleShape *shape)
+{
+  auto input_node = loco::must_cast<luci::CircleNode *>(shape->input());
+  auto name = input_node->name();
+  assert(name.length() > 0);
+  auto shape_status = input_node->shape_status();
+  if (shape_status != luci::ShapeStatus::VALID)
+    return nullptr;
+  auto rank = input_node->rank();
+  if (rank == 0)
+    return nullptr;
```

Comment:
let's check this how TF or pytorch act.

Reply:
```cxx
  auto rank = input_node->rank();
  if (rank == 0)
    return nullptr;
```cxx
->
```
  auto rank = input_node->rank();
  // support rank == 0 when possible
  if (rank == 0)
    return nullptr;
```

Reply:
> // support rank == 0 when possible

I understood this comment as a `TODO` item.. am I right?

Reply:
yup